packages/siop-oid4vp/lib/authorization-response/Dcql.ts

```diff
@@ -0,0 +1,59 @@
+import { Hasher } from '@sphereon/ssi-types'
+import { DcqlMdocRepresentation, DcqlPresentationRecord, DcqlQuery, DcqlSdJwtVcRepresentation } from 'dcql'
+import { DcqlPresentationQueryResult } from 'dcql'
+
+import { extractDataFromPath } from '../helpers'
+import { AuthorizationRequestPayload, SIOPErrors } from '../types'
+
+import { extractPresentationRecordFromDcqlVpToken } from './OpenID4VP'
+
+/**
+ * Finds a valid DcqlQuery inside the given AuthenticationRequestPayload
+ * throws exception if the DcqlQuery is not valid
+ * returns the decoded dcql query if a valid instance found
+ * @param authorizationRequestPayload object that can have a dcql_query inside
+ * @param version
+ */
+export const findValidDcqlQuery = async (authorizationRequestPayload: AuthorizationRequestPayload): Promise<DcqlQuery | undefined> => {
+  const dcqlQuery: string[] = extractDataFromPath(authorizationRequestPayload, '$.dcql_query').map((d) => d.value)
+  const definitions = extractDataFromPath(authorizationRequestPayload, '$.presentation_definition')
+  const definitionsFromList = extractDataFromPath(authorizationRequestPayload, '$.presentation_definition[*]')
+  const definitionRefs = extractDataFromPath(authorizationRequestPayload, '$.presentation_definition_uri')
+  const definitionRefsFromList = extractDataFromPath(authorizationRequestPayload, '$.presentation_definition_uri[*]')
```

Comment:
Is it really needed to use json path for this? Why not just do `authorizationRequestPayload.presentation_definition`, `authorizationRequestPayload.presentation_definition_uri`, etc.??

Reply:
Because it's done in this way for pex as well

Reply:
dcql wanted to get rid of jsonpath :P

Reply:
Reason was/is that in older versions of the spec these objects could also live elsewhere btw